src/ai/defaultai.cc

```diff
@@ -981,6 +975,12 @@ void DefaultAI::late_initialization() {
 					bo.critical_building_material.push_back(temp_buildcosts.first);
 				}
 			}
+			for (const auto& temp_buildcosts : train.enhancement_cost()) {
```

Comment:
here we fill the list of critical buildmaterial for trainingsite enhancements which we need in defaultai_warfare.cc